common/meta/raft-store/src/state_machine/sm.rs

```diff
@@ -559,6 +576,265 @@ impl StateMachine {
+                if cond.target == ConditionTarget::Value as i32 {
+                    // compare the value
+                    self.return_value_condition_result(cond.expected, cond, sv)
+                } else {
+                    // compare the seq
+                    self.return_seq_condition_result(cond.expected, cond, sv)
+                }
+            }
+            None => false,
+        }
+    }
+
+    #[tracing::instrument(level = "debug", skip(self, txn_tree))]
+    fn txn_execute_condition(
+        &self,
+        txn_tree: &TransactionSledTree,
+        condition: &Vec<TxnCondition>,
+    ) -> bool {
+        for cond in condition {
+            if !self.txn_execute_one_condition(txn_tree, cond) {
+                return false;
+            }
+        }
+
+        true
+    }
+
+    fn txn_execute_get_operation(
+        &self,
+        txn_tree: &TransactionSledTree,
+        get: &TxnGetRequest,
+        resp: &mut TxnReply,
+    ) {
+        let key = get.key.clone();
+
+        let sub_tree = txn_tree.key_space::<GenericKV>();
+        let sv = sub_tree.get(&key);
+
+        let get = match sv {
+            Ok(v) => match v {
+                Some(v) => TxnGetResponse {
+                    key: get.key.clone(),
+                    value: convert_seqv_to_pb(Some(v)),
+                },
+                None => TxnGetResponse {
+                    key: get.key.clone(),
+                    value: None,
+                },
+            },
+            Err(_v) => TxnGetResponse {
+                key: get.key.clone(),
+                value: None,
+            },
+        };
+
+        resp.responses.push(TxnOpResponse {
+            response: Some(txn_op_response::Response::Get(get)),
+        });
+    }
+
+    fn txn_execute_put_operation(
+        &self,
+        txn_tree: &TransactionSledTree,
+        put: &TxnPutRequest,
+        resp: &mut TxnReply,
+    ) {
+        let key = put.key.clone();
+        let sub_tree = txn_tree.key_space::<GenericKV>();
+        let key_str = key.to_string();
+
+        let result = self.txn_sub_tree_upsert(
+            &sub_tree,
+            &key_str,
+            &MatchSeq::Any,
+            Operation::Update(put.value.clone()),
+            None,
+        );
+
+        let put_resp = match result {
+            Ok((prev, result)) => {
+                tracing::debug!("applied txn_execute_put_operation: {} {:?}", key, result);
+
+                TxnPutResponse {
+                    key: put.key.clone(),
+                    prev_value: if put.prev_kv {
+                        convert_seqv_to_pb(prev)
+                    } else {
+                        None
+                    },
+                }
+            }
+            Err(err) => {
+                tracing::debug!("txn_execute_put_operation fail: {} {:?}", key, err);
+                TxnPutResponse {
+                    key: put.key.clone(),
+                    prev_value: None,
+                }
+            }
```

Comment:
Just logging an error is not enough. Storage level error has to be dealt with.

Reply:
done